Utility/Framework/SymfonyFramework.php

```diff
@@ -75,19 +78,6 @@ public static function isKernelLessThan($major, $minor = null, $patch = null)
      */
     private static function kernelVersionCompare($operator, $major, $minor = null, $patch = null)
     {
-        $vernum = $major;
-        $kernel = Kernel::MAJOR_VERSION;
-
-        if ($minor) {
-            $vernum .= '.'.$minor;
-            $kernel .= '.'.Kernel::MINOR_VERSION;
-
-            if ($patch) {
-                $vernum .= '.'.$patch;
-                $kernel .= '.'.Kernel::RELEASE_VERSION;
-            }
-        }
-
-        return version_compare($kernel, $vernum, $operator);
+        return version_compare(Kernel::VERSION_ID, sprintf("%d%'.02d%'.02d", $major, $minor ?: 0, $patch ?: 0), $operator);
```

Comment:
Yeah, feature detection is of course preferable. Thankfully, this class is no longer required (for either version or feature detection) on the `2.x` branch. You may also feel better (as I do) that version detection is only used once or twice in the production codebase; the majority of calls occur in the test suite.

:-)